modules/battery_manager/php/testprovisioner.class.inc

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\battery_manager;
+
+/**
+ * This class implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class TestProvisioner extends \LORIS\Data\ProvisionerInstance
+{
+    /**
+     * Select all tests from the test_battery table that match the attribute
+     * values passed as conditions and will return an array of test objects
+     * indexed by id
+     *
+     * @return array $tests     List of Test Objects that match the query
+     */
+    public function selectInstances() : array
+    {
+        $db    = \Database::singleton();
+        $query = "SELECT b.id,
+                          t.Test_name as testName,
+                          b.AgeMinDays as ageMinDays,
+                          b.AgeMaxDays as ageMaxDays,
+                          b.Stage as stage,
+                          s.SubprojectID as subproject,
+                          b.Visit_label as visitLabel,
+                          p.CenterID as centerId,
+                          b.firstVisit,
+                          b.instr_order as instrumentOrder,
+                          b.Active as active
+                   FROM test_battery b
+                     LEFT JOIN test_names t USING (Test_name)
+                     LEFT JOIN subproject s USING (SubprojectID)
+                     LEFT JOIN psc p USING (CenterID)
+                   ORDER BY Full_name";
+
+        $result = $db->pselect($query, array());
+        $tests  = array();
+        if (!empty($result)) {
+            foreach ($result as $row) {
+                $tests[$row['id']] = $row;
+            }
+        }
+
+        return $tests;
+    }
+
+    /**
+     * Returns an instance of a Test object for a given
+     * table row.
+     *
+     * @param array $row The database row from the LORIS Database class.
+     *
+     * @return \LORIS\Data\DataInstance An instance representing a test.
+     */
+    public function getInstance($row) : \LORIS\Data\DataInstance
+    {
+        $cid = $row['centerId'];
+        return new TestIntance($row, $cid);
+    }
+
+    /**
+     * Instantiates an ArrayIterator class that is composed of all the Test
+     * Objects
```

Comment:
Some function description end with a dot while other without. Should try to use the same ending.